test/test_conversion.py

```diff
@@ -270,3 +270,13 @@ class HAVCMode(enum.Enum):
         self.assertEqual(bytes([2]), knxdclient.encode_value(HAVCMode.Standby, knxdclient.KNXDPT.ENUM8))
         self.assertEqual(bytes([2]), knxdclient.encode_value(2, knxdclient.KNXDPT.ENUM8))
         self.assertEqual(2, knxdclient.decode_value(bytes([2]), knxdclient.KNXDPT.ENUM8))
+
+    def test_dpt232_conversion(self) -> None:
+        self.assertEqual(bytes([0x10, 0xCC, 0x0A]),
+                         knxdclient.encode_value(bytes([0x10, 0xCC, 0x0A]), knxdclient.KNXDPT.COLOUR_RGB))
+        self.assertEqual(bytes([127, 00, 32]),
+                         knxdclient.encode_value(bytes([0x7F, 0x00, 0x20]), knxdclient.KNXDPT.COLOUR_RGB))
+        self.assertEqual(f'{0x801B10:X}',
+                         knxdclient.decode_value(bytes([0x80, 0x1B, 0x10]), knxdclient.KNXDPT.COLOUR_RGB))
+        self.assertEqual(f'{0x123456:X}',
+                         knxdclient.decode_value(bytes([0x12, 0x34, 0x56]), knxdclient.KNXDPT.COLOUR_RGB))
```

Comment:
You could add a "roundtrip" test here that checks that `decode_value(encode_value(some_value, ...), ...) == some_value`.

This should also reveal the issue with the deviating datatypes, discussed in my other comments.

Reply:
Ta. I shall tinker. (This first one might take us a little while, but it should speed any future ones. I appreciate the hand-holding).